File: picker/src/main/java/com/picker8/model/PhotoDirectory.java
package com.picker8.model;

import android.text.TextUtils;

import com.hy.picker.utils.SetList;

/**
 * Created by donglua on 15/6/28.
 */
public class PhotoDirectory {

    private String id;
    private String coverPath;
    private String name;
    private long dateAdded;
    private SetList<Photo> photos = new SetList<>();

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof PhotoDirectory)) return false;

        PhotoDirectory directory = (PhotoDirectory) o;

        boolean hasId = !TextUtils.isEmpty(id);
        boolean otherHasId = !TextUtils.isEmpty(directory.id);

        if (hasId && otherHasId) {
            return TextUtils.equals(id, directory.id) && TextUtils.equals(name, directory.name);

        }

        return false;
    }

    @Override
    public int hashCode() {
        if (TextUtils.isEmpty(id)) {
            if (TextUtils.isEmpty(name)) {
                return 0;
            }

            return name.hashCode();
        }

        int result = id.hashCode();

        if (TextUtils.isEmpty(name)) {
            return result;
        }

        result = 31 * result + name.hashCode();
        return result;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getCoverPath() {
        return coverPath;
    }

    public void setCoverPath(String coverPath) {
        this.coverPath = coverPath;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getDateAdded() {
        return dateAdded;
    }

    public void setDateAdded(long dateAdded) {
        this.dateAdded = dateAdded;
    }

    public SetList<Photo> getPhotos() {
        return photos;
    }

    public void setPhotos(SetList<Photo> photos) {
        this.photos = photos;
    }

    public void addPhoto(Photo photo) {
        photos.add(photo);
    }

}

File: picker/src/main/java/com/hy/picker/model/PhotoDirectory.kt
package com.hy.picker.model

import android.text.TextUtils

/**
 * Created by donglua on 15/6/28.
 */
class PhotoDirectory {

    var id: String? = null
    var coverPath: String? = null
    var name: String? = null
    var dateAdded = 0L
    val photos = ArrayList<Photo>()

    fun addPhoto(photo: Photo) {
        photos.add(photo)
    }

    fun addPhoto(index: Int, photo: Photo) {
        photos.add(index, photo)
    }


    override fun equals(other: Any?): Boolean {
        if (this === other) return true

        if (javaClass != other?.javaClass) return false

        other as PhotoDirectory

        val hasId = !TextUtils.isEmpty(id)
        val otherHasId = !TextUtils.isEmpty(other.id)

        if (hasId && otherHasId) {
            return id == other.id && name == other.name
        }

        return false
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (name?.hashCode() ?: 0)
        return result
    }

}
